lib/rules/sort-keys.js

```diff
@@ -159,10 +167,40 @@ module.exports = {
                 const numKeys = stack.numKeys;
                 const thisName = getPropertyName(node);
 
+                // Get tokens between current node and previous node
+                const tokens = stack.prevNode && sourceCode
+                    .getFirstTokensBetween(stack.prevNode, node, { skip: 0, includeComments: true });
+
+                let isBlankLineBetweenNodes = stack.prevBlankLine;
+
+                if (tokens) {
+
+                    // check blank line between tokens
+                    tokens.forEach((token, index) => {
+                        const previousToken = tokens[index - 1];
+
+                        if (previousToken && (token.loc.start.line - previousToken.loc.end.line > 1)) {
+                            isBlankLineBetweenNodes = true;
+                        }
+                    });
+
+                    // check blank line between current node and last token
+                    if (!isBlankLineBetweenNodes && (node.loc.start.line - tokens[tokens.length - 1].loc.end.line > 1)) {
+                        isBlankLineBetweenNodes = true;
+                    }
+                }
```

Comment:
I think this should also count as line separated groups (this is [comma-first style](https://eslint.org/docs/latest/rules/comma-style#first)):

```js
/* eslint sort-keys: ["error", "asc", { allowLineSeparatedGroups: true } ] */

var obj = {
  b

  ,a // false positive
}

```

What's missing in the checks is comparing the start line of the first token between nodes (comma in this example) with the end line of the previous node (`b` in this example).

Reply:
Added this check too - https://github.com/eslint/eslint/pull/16138/commits/4c2ba24a21ac7091b6002bac4191bbf61a74cc2f